commit 70b3fc3bb866f8d5f5a71d42ddb5486cbf89f2ed
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Fri Aug 13 09:24:04 2010 -0700

    mm: fix missing page table unmap for stack guard page failure case
    
    .. which didn't show up in my tests because it's a no-op on x86-64 and
    most other architectures.  But we enter the function with the last-level
    page table mapped, and should unmap it at exit.
    
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/mm/memory.c b/mm/memory.c
index a1a8e45..659776b 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -2309,8 +2309,10 @@ static int do_anonymous_page(struct mm_struct *mm, struct vm_area_struct *vma,
 	spinlock_t *ptl;
 	pte_t entry;
 
-	if (check_stack_guard_page(vma, address) < 0)
+	if (check_stack_guard_page(vma, address) < 0) {
+		pte_unmap(page_table);
 		return VM_FAULT_SIGBUS;
+	}
 
 	/* Allocate our own private page. */
 	pte_unmap(page_table);
